### PR TITLE
feat(License): add new headers information about license

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
## Changes Proposed:
- Added new file `AUTHORS` for copyright information
- Change all headers for new copyright information
- Added missing license info for few files

### CODE REVIEW

**Review by Helias — APPROVED** [review]

**Comment by Helias** [comment]
@azerothcore/developers 

@FrancescoBorzi @Yehonal @BarbzYHOOL

File: AUTHORS [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+* github.com/azerothcore/azerothcore-wotlk
+
+Unfortunately we have no detailed information of the history of the WoWD project;
+if somebody can provide information, please contact us, so that we can make this history available
+
+## Exceptions with third party libraries
+The third party libraries have their own way of addressing authorship, and the authorship of commits importing/ updating
+a third party library reflects who did the importing instead of who wrote the code within the commit.
+
+The Authors of third party libraries are not explicitly mentioned, and usually are possible to obtain from the files belonging to the third party libraries.
```
Comment (Kitzunu) [inline AUTHORS:35]: newlien

File: AUTHORS [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+## History of development
+Development of this project dates back to 2004, and was developed under various umbrellas over time:
+* WoW Daemon Team, 2004
+* MaNGOS project, 2005-2008, located at http://www.mangosproject.org
+* MaNGOS project, 2008-2011, located at http://getmangos.com
+* SD2 project, 2008-2009, located at http://getmangos.com
+* TrinityCore, 2008-2021, located at https://www.trinitycore.org/
+* SunwellCore, located at https://www.azerothcore.org/pages/sunwell.pl/
+* AzerothCore, 2016-2021, located at https://www.azerothcore.org/
```
Comment (Kitzunu) [inline AUTHORS:20]: Add Warhead?

**Review by Kitzunu — APPROVED** [review]

**Comment by Kitzunu** [comment]
Nice changes, I like it

File: AUTHORS (hunk 2 above)
Comment (Winfidonarleyan) [reply AUTHORS:20]: WarheadCore based ac core

**Comment by Winfidonarleyan** [comment]
@Yehonal can check this?

File: AUTHORS [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# List of AUTHORS who contributed over time to the AzerothCore project
+
+## Warning
+The code of AzerothCore is shipped as it is without any form of warranty,
+and - except for third party libraries licensed
+- under the GPL 2 for old files which you can read from the file "LICENSE-GPL2"
+- under the AGPL 3 for new files which you can read from the file "LICENSE-AGPL3"
```
Comment (FrancescoBorzi) [inline AUTHORS:7]: I think we can get rid of the double licensing as all GPLv2 source can be released with "GPLv2 or latter" version

File: AUTHORS [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+## Point of current development
+The project is currently hosted at https://www.azerothcore.org/ and developed under https://github.com/azerothcore
+
+## History of development
+Development of this project dates back to 2004, and was developed under various umbrellas over time:
+* WoW Daemon Team, 2004
+* MaNGOS project, 2005-2008, located at http://www.mangosproject.org
+* MaNGOS project, 2008-2011, located at http://getmangos.com
+* SD2 project, 2008-2009, located at http://getmangos.com
```
Comment (FrancescoBorzi) [inline AUTHORS:17]: I'd add UDB too, also SD2 wasn't located at getmangos.com but at www.scriptdev2.com iirc

File: AUTHORS [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+* TrinityCore, 2008-2021, located at https://www.trinitycore.org/
+* SunwellCore, located at https://www.azerothcore.org/pages/sunwell.pl/
```
Comment (FrancescoBorzi) [inline AUTHORS:19]:
```suggestion
* TrinityCore, 2008-2012, located at https://www.trinitycore.org/
* SunwellCore 2012-2016, privately developed, more info at https://www.azerothcore.org/pages/sunwell.pl/
```

File: AUTHORS [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+* AzerothCore, 2016-2021, located at https://www.azerothcore.org/
+
+## Authorship of the code
+Authorship is assigned for each commit within the git history, which is stored in these git repositories:
+* github.com/cmangos/mangos-svn (History from MaNGOS project from 2005-2008, originally hosted at http://mangos.svn.sourceforge.net)
+* github.com/TrinityCore/TrinityCore
+* github.com/azerothcore/azerothcore-wotlk
+
+Unfortunately we have no detailed information of the history of the WoWD project;
+if somebody can provide information, please contact us, so that we can make this history available
```
Comment (FrancescoBorzi) [inline AUTHORS:29]: we should also underline how there is no git history of sunwellcore

File: AUTHORS (hunk 3 above)
Comment (Winfidonarleyan) [reply AUTHORS:7]: Okay

File: AUTHORS (hunk 4 above)
Comment (Winfidonarleyan) [reply AUTHORS]: > I'd add UDB too
I do not know what it is :/

File: AUTHORS (hunk 6 above)
Comment (Winfidonarleyan) [reply AUTHORS:28]: Added, need check

File: AUTHORS (hunk 4 above)
Comment (Winfidonarleyan) [reply AUTHORS]: Replace sd url to http://www.scriptdev2.com/

File: AUTHORS (hunk 3 above)
Comment (Winfidonarleyan) [reply AUTHORS]: Updated

File: AUTHORS (hunk 6 above)
Comment (Kitzunu) [reply AUTHORS:28]: Maybe we can just add Xinef and Pussywizzard for swc. That way we get some info about it

**Comment by Winfidonarleyan** [comment]
Can merge? If need add info, we can do it later.

**Review by Yehonal — APPROVED** [review]

**Comment by Kitzunu** [comment]
Good

**Comment by Winfidonarleyan** [comment]
> Good

Well of course

**Pull request merged**